components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/datasource/QueryExecutor.java

```diff
@@ -111,12 +110,13 @@ public boolean executeUpdateBusinessRuleQuery(String uuid, byte[] newBusinessRul
         PreparedStatement statement = null;
         try {
             conn = dataSource.getConnection();
-            conn.setAutoCommit(true);
+            conn.setAutoCommit(false);
             statement = getUpdateBusinessRuleQuery(conn, uuid, newBusinessRule, deploymentStatus);
             if (statement == null) {
                 return false;
             }
             result = statement.execute();
+            conn.commit();
             return result;
         } catch (SQLException e) {
```

Comment:
have the stack trace. Fix other places if needed

Reply:
Fixed by 2c0119c4c48e1077512e685493252c388e68a520